core/src/main/java/hudson/Util.java

```diff
@@ -367,10 +363,8 @@ public static boolean isSymlink(@Nonnull File file) throws IOException {
     @SuppressWarnings("NP_BOOLEAN_RETURN_NULL")
     private static Boolean isSymlinkJava7(@Nonnull File file) throws IOException {
         try {
-            Object path = File.class.getMethod("toPath").invoke(file);
-            return (Boolean) Class.forName("java.nio.file.Files").getMethod("isSymbolicLink", Class.forName("java.nio.file.Path")).invoke(null, path);
-        } catch (NoSuchMethodException x) {
-            return null; // fine, Java 6
+            Path path = file.toPath();
+            return Files.isSymbolicLink(path);
         } catch (Exception x) {
```

Comment:
[`Files.isSymbolicLink(Path)`](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#isSymbolicLink(java.nio.file.Path)) only declares that it throws a `SecurityException`.  So I was wondering if

* this catch should be to catch `SecurityException` only
* and at the same time there are many places where we do not catch `SecurityException` and just let it ripple out.  I mean if `checkRead` fails on this file - then it would seem like Jenkins has bigger issues?